### PR TITLE
Lit Cakes (modular + tested)

### PR DESCRIPTION
**Pull request by Odairu**
## About The Pull Request
Adds the ability for Birthday Cakes to emit light similar to a red candle via any fire source, as well as be extinguished via fire extinguisher

## Why It's Good For The Game
My birthday cake can now actually light up my dark room

## Changelog

:cl:
add: Birthday Cakes are now able to be lit and emit light

/:cl:

### CODE REVIEW

**Review by projectkepler-ru — APPROVED** [review]

**Comment by Majkl-J** [comment]
As a note to headtainers: Keep this on hold for about 2-3 days, I want to try my shot at optimizing this

File: modular_zubbers/modules/modular_items/code/cake_light.dm [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+// allows lighting an unlit candle from some fire source by left clicking the source with the candle
+/obj/item/food/cake/birthday/pre_attack(atom/target, mob/living/user, params)
+	if(ismob(target))
+		return ..()
+
+	if(try_light_candle(target, user, quiet = TRUE))
+		return COMPONENT_CANCEL_ATTACK_CHAIN
+
+	return ..()
```
Comment (Majkl-J) [inline modular_zubbers/modules/modular_items/code/cake_light.dm:85]:
```suggestion
#define FAILURE 0
#define SUCCESS 1
#define ALREADY_LIT 3

/obj/item/flashlight/flare/candle/cake

/obj/item/food/cake/birthday
	var/obj/item/flashlight/flare/candle/cake/candle

/obj/item/food/cake/birthday/Initialize(mapload)
	. = ..()
	candle = new(src)

/obj/item/food/cake/birthday/extinguish()
	. = ..()
	candle.extinguish()

/obj/item/food/cake/birthday/fire_act(exposed_temperature, exposed_volume)
	candle.fire_act(exposed_temperature, exposed_volume)
	return ..()

/obj/item/food/cake/birthday/attackby(obj/item/attacking_item, mob/user, params)
	candle.attackby(attacking_item, user, params)
	return ..()

// allows lighting an unlit candle from some fire source by left clicking the source with the candle
/obj/item/food/cake/birthday/pre_attack(atom/target, mob/living/user, params)
	candle.pre_attack(target, user, params)
	return ..()
```

**Review by Majkl-J — CHANGES_REQUESTED** [review]
Less copy paste code

File: modular_zubbers/modules/modular_items/code/cake_light.dm (hunk 1 above)
Comment (Majkl-J) [reply modular_zubbers/modules/modular_items/code/cake_light.dm:85]: You can just click the commit suggestion

**Review by Majkl-J — APPROVED** [review]
Merge ready

**Pull request merged**